### PR TITLE
feat: add udp port scan

### PR DESCRIPTION
**Pull request by henok-tesfaye**
A feature PR which adds UDP port scan.

### CODE REVIEW

File: portscanner.py [hunk 1]
```diff
@@ -51,6 +53,21 @@ def tcp_threader():
         port = q.get()
         tcp_scan(port)
         q.task_done()
+        
+def udp_scan(port):
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        s.sendto('invalid request', (host, port))
```
Comment (mezzzi) [inline portscanner.py:60]: does this work? I thought you said udp scanning was too complicated

Comment (henok-outlier) [reply portscanner.py:60]: With this code, we can detect unclosed ports(that doesn't mean ports are opened. Some packets may be rejected by firewall).

Comment (henok-outlier) [reply portscanner.py:60]: UDP servers won't reply for requests that don't understand, but if a port is closed the OS will trigger an exception.

So currently we can only detect closed ports, and assumed all ports which are not closed as open.

File: portscanner.py [hunk 2]
```diff
@@ -51,6 +53,21 @@ def tcp_threader():
...
+        with print_lock:
+            print('%-15s%-15s%-15s' % (str(port) +'/udp', 'open', getPortService(port, 'udp')))
```
Comment (mezzzi) [inline portscanner.py]: So should the port status be `open` or something else, like `unclosed` or `possibly-open`?

Comment (henok-tesfaye) [reply portscanner.py]: right, how does `open | filter` sound?

Comment (mezzzi) [reply portscanner.py]: what is the `filter` for?

Comment (henok-outlier) [reply portscanner.py]: for packets rejected by firewall

Comment (mezzzi) [reply portscanner.py]: Ok, make it `filtered` instead of `filter`

Comment (henok-tesfaye) [reply portscanner.py]: will do that

**Review by mezzzi — APPROVED** [review]

**Pull request merged**